marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/EpochStakepoolSize.hs

```diff
@@ -113,37 +118,63 @@ test = H.integration . HE.runFinallies . TN.workspace "chairman" $ \tempAbsPath
   -- Prepare transaction to register stakepool and stake funds
   (poolVKey :: C.PoolId, tx, txBody) <- registerPool con networkId pparams tempAbsPath keyWitnesses [stakeCredential, stakeCredential2] genesisAddress
 
-  -- start indexer
-  found <- liftIO IO.newEmptyMVar
-  let dbPath = tempAbsPath </> "epoch_stakepool_sizes.db"
-  dbCon <- liftIO $ SQL.open dbPath
-  void $ liftIO $ do
-    chan <- IO.newChan
-    let indexer = CS.ledgerStates (TN.configurationFile runtime) socketPath C.QuickValidation
-          & EpochStakepoolSize.toEvents
-          & EpochStakepoolSize.sqlite dbCon
-          & S.chain (IO.writeChan chan) -- After indexer has written the event to database, we write it to the chan
-    void $ (IO.link =<<) $ IO.async $ void $ S.effects indexer
-
-    -- Consume the channel until an event is found which (1) has the
-    -- pool ID and (2) has the right amount of lovelace staked.
-    (IO.link =<<) $ IO.async $ forever $ do
-      EpochStakepoolSize.Event (_epochNo, stakeMap) <- IO.readChan chan
-      case Map.lookup poolVKey stakeMap of
-        Just lovelace -> when (lovelace == totalStakedLovelace) $ IO.putMVar found () -- Event found!
-        _             -> return ()
-
   -- Submit transaction to create stakepool and stake the funds
   TN.submitAwaitTx con (tx, txBody)
 
+  -- This is the channel we wait on to know if the event has been indexed
+  indexedTxs <- liftIO IO.newChan
+  -- Start indexer
+  coordinator <- liftIO $ Indexers.initialCoordinator 1
+  ch <- liftIO $ STM.atomically . STM.dupTChan $ Indexers._channel coordinator
+  let dbPath = tempAbsPath </> "epoch_stakepool_sizes.db"
+  (loop, _indexerMVar) <- liftIO $ Indexers.epochStakepoolSizeWorker_
+      (TN.configurationFile runtime)
+      (STM.writeChan indexedTxs)
+      123
```

Comment:
The correct security param for the testnets appears to be 10.

Reply:
8f04c23774d0ce46ea5e6b5ba5350163851ca66c